Test Failure: libraries pgo assert in outerloop

System.Runtime.Tests assertion failure in libraries-pgo, outerloop
Fails on Linux+Windows x64

```
Assert failure(PID 2736 [0x00000ab0], Thread: 4260 [0x10a4]): Assertion failed 'varDsc->CanBeReplacedWithItsField(m_compiler) || varDsc->lvDoNotEnregister || !varDsc->lvPromoted' in 'System.Tests.UInt128Tests_GenericMath:DivideModeTest()' during 'Lowering nodeinfo' (IL size 957; hash 0x0e5ed17e; Tier1-OSR)
```

Helix link
https://helix.dot.net/api/2019-06-17/jobs/51fdb3d4-8e62-41d8-b6a7-a9981159cfd5/workitems/System.Runtime.Tests/console

Failing run:
https://dev.azure.com/dnceng-public/public/_build/results?buildId=1386172

Native AOT outerloops are now also broken:

```
2026-05-21T09:37:05.1597375Z     ILC: D:\a\_work\1\s\src\coreclr\jit\lower.cpp:5605
2026-05-21T09:37:05.1639917Z     ILC: Assertion failed 'varDsc->CanBeReplacedWithItsField(m_compiler) || varDsc->lvDoNotEnregister || !varDsc->lvPromoted' in 'JIT.HardwareIntrinsics.X86.BigMulTestSuite:TestLargeRandomCalculations()' during 'Lowering nodeinfo' (IL size 150; hash 0x426b9eb9; FullOpts)
2026-05-21T09:37:05.1640844Z     
2026-05-21T09:37:06.2416219Z ##[error]artifacts\bin\coreclr\windows.x64.Checked\build\Microsoft.NETCore.Native.targets(311,5): error MSB3073: The command ""D:\a\_work\1\s\artifacts\bin\coreclr\windows.x64.Checked\x64\ilc\\ilc" @"D:\a\_work\1\s\artifacts\tests\coreclr/obj/windows.x64.Checked/Managed/JIT\HardwareIntrinsics\HardwareIntrinsics_X86_r\native\HardwareIntrinsics_X86_r.ilc.rsp"" exited with code 57005. [D:\a\_work\1\s\src\tests\JIT\HardwareIntrinsics\HardwareIntrinsics_X86_r.csproj]
```

https://dev.azure.com/dnceng-public/public/_build/results?buildId=1429516&view=logs&jobId=0abb9308-48e7-575c-35a3-0d90d6fb0888&j=0abb9308-48e7-575c-35a3-0d90d6fb0888&t=b2f3a958-309d-5f1d-3cb7-eab59ea182cc

Is there an ETA?

We'd need to disable the entire HardwareIntrinsics_X86_r.csproj project on native AOT since this is a build break.

--- a/src/coreclr/jit/optcse.cpp
+++ b/src/coreclr/jit/optcse.cpp
@@ -1803,7 +1803,9 @@ bool CSE_HeuristicCommon::CanConsiderTree(GenTree* tree, bool isReturn)
         // IR that lower is not yet prepared to handle.
         //
         if (isReturn)
+        {
             return false;
+        }
 
         // The locals we introduce cannot be be enregistered in multiple
         // registers since we do not promote them, so they would always be
@@ -1811,7 +1813,9 @@ bool CSE_HeuristicCommon::CanConsiderTree(GenTree* tree, bool isReturn)
         // destinations when replacing the CSEs uses and we do not currently do
         // that.
         if (tree->IsMultiRegNode())
+        {
             return false;
+        }
     }
 
     // No good if the expression contains side effects or if it was marked as DONT CSE

--- a/src/coreclr/jit/optcse.cpp
+++ b/src/coreclr/jit/optcse.cpp
@@ -1796,13 +1796,22 @@ bool CSE_HeuristicCommon::CanConsiderTree(GenTree* tree, bool isReturn)
         }
     }
 
-    // Don't allow non-SIMD struct CSEs under a return; we don't fully
-    // re-morph these if we introduce a CSE store, and so may create
-    // IR that lower is not yet prepared to handle.
-    //
-    if (isReturn && varTypeIsStruct(tree->gtType) && !varTypeIsSIMD(tree->gtType))
+    if (varTypeIsStruct(tree->gtType) && !varTypeIsSIMD(tree->gtType))
     {
-        return false;
+        // Don't allow non-SIMD struct CSEs under a return; we don't fully
+        // re-morph these if we introduce a CSE store, and so may create
+        // IR that lower is not yet prepared to handle.
+        //
+        if (isReturn)
+            return false;
+
+        // The locals we introduce cannot be be enregistered in multiple
+        // registers since we do not promote them, so they would always be
+        // spilled. Also, for correctness we would need to DNER existing store
+        // destinations when replacing the CSEs uses and we do not currently do
+        // that.
+        if (tree->IsMultiRegNode())
+            return false;
     }
 
     // No good if the expression contains side effects or if it was marked as DONT CSE

--- a/src/coreclr/jit/optcse.cpp
+++ b/src/coreclr/jit/optcse.cpp
@@ -1807,11 +1807,12 @@ bool CSE_HeuristicCommon::CanConsiderTree(GenTree* tree, bool isReturn)
             return false;
         }
 
-        // The locals we introduce cannot be be enregistered in multiple
-        // registers since we do not promote them, so they would always be
-        // spilled. Also, for correctness we would need to DNER existing store
-        // destinations when replacing the CSEs uses and we do not currently do
-        // that.
+        // Skip all multireg nodes. The locals we introduce cannot be
+        // enregistered in multiple registers since we do not promote them, so
+        // they would always be spilled. Also, for correctness we would need to
+        // DNER existing store destinations when replacing the CSE uses and we
+        // do not currently do that.
+        //
         if (tree->IsMultiRegNode())
         {
             return false;

